pkg/controller/master/addon/patch_grafana.go

```diff
@@ -37,7 +37,7 @@ func (h *Handler) MonitorAddonRancherMonitoring(key string, aObj *harvesterv1.Ad
 	}
 
 	// disabled successfully, the pv has already from Bound to Released/Available
-	if !aObj.Spec.Enabled && aObj.Status.Status == "" {
+	if !aObj.Spec.Enabled && aObj.Status.Status == harvesterv1.AddonInitState {
```

Comment:
When I switch to enable and disable rancher monitoring, it will create a new PV.
I think the addon status misses an init status.
<img width="1706" alt="image" src="https://github.com/harvester/harvester/assets/3347460/e7641e74-5caa-44d4-8758-138c7d4ed479">

Reply:
this is being caused due to install job being terminated by the helm controller when job is being patched.

as a result the pv gets used with the pvc claim but doesnt get cleaned up when `helm install` is terminated by helm controller mid way due to the patched job.

I have moved the patching logic into the controller as it cant be done later and doesnt neccessarily need to be done by a mutating webhook. This ensures that the helm install job is not terminated and pv is correctly used. 

the latest commit has the fixes.